File: Bit Manipulation/0191/0191.cpp
#include <iostream>

class Solution
{
private:
    /* data */
public:
    int func_191(int data, int bits=4);
    Solution(/* args */);
    ~Solution();
};

Solution::Solution(/* args */)
{
}

Solution::~Solution()
{
}

int Solution::func_191(int data, int bits)
{
    int result = 0;
    int mask = 1;
    for(int i=0; i<32; i++){
        if ((mask & data) != 0)
            result += 1; // / mask;
        mask = mask << 1;
    }
    return result;
}

int main(int argc, char** argv){
    int data = 11;
    Solution ss;
    std::cout << ss.func_191(data) << std::endl;    
    return 0;
}

File: Bit Manipulation/0191/0191.py
# 和190类似，解码每个二进制位
def func_191(data):
    r = 0
    for i in range(32):
        r += data % 2
        data = data // 2
        if data == 0:
            break
    return r

#  移动掩膜
def hammingWeight(self, n: int) -> int:
    count = 0
    
    mask = 1
    for _ in range(32):
        if n & mask != 0:
            count += 1
        mask = mask << 1
        
    return count
data = 11  # 1011
print(func_191(data))  # 3